storm-core/src/jvm/backtype/storm/blobstore/LocalFsBlobStore.java

```diff
@@ -0,0 +1,306 @@
+    BlobStoreFileOutputStream mOut = null;
+    try {
+      mOut = new BlobStoreFileOutputStream(fbs.write(META_PREFIX+key, false));
+      mOut.write(Utils.thriftSerialize((TBase) meta));
+      mOut.close();
+      mOut = null;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } finally {
+      if (mOut != null) {
+        try {
+          mOut.cancel();
+        } catch (IOException e) {
+          //Ignored
+        }
+      }
+    }
+  }
+
+  @Override
+  public void deleteBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException {
+    validateKey(key);
+    checkForBlobOrDownload(key);
+    SettableBlobMeta meta = getStoredBlobMeta(key);
+    _aclHandler.hasPermissions(meta.get_acl(), WRITE, who, key);
+    try {
+      fbs.deleteKey(DATA_PREFIX+key);
+      fbs.deleteKey(META_PREFIX+key);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public InputStreamWithMeta getBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException {
+    validateKey(key);
+    if(!checkForBlobOrDownload(key)) {
+      checkForBlobUpdate(key);
+    }
+    SettableBlobMeta meta = getStoredBlobMeta(key);
+    _aclHandler.hasPermissions(meta.get_acl(), READ, who, key);
+    try {
+      return new BlobStoreFileInputStream(fbs.read(DATA_PREFIX+key));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Iterator<String> listKeys(Subject who) {
+    try {
+      return new KeyTranslationIterator(fbs.listKeys(), DATA_PREFIX);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void shutdown() {
+  }
+
+  @Override
+  public int getBlobReplication(String key, Subject who) throws Exception {
+    validateKey(key);
+    SettableBlobMeta meta = getStoredBlobMeta(key);
+    _aclHandler.hasPermissions(meta.get_acl(), READ, who, key);
+    CuratorFramework zkClient = Utils.createZKClient(conf);
+    if (zkClient.checkExists().forPath(BLOBSTORE_SUBTREE + key) == null) {
+       zkClient.close();
+       return 0;
+    }
+    int replicationCount = zkClient.getChildren().forPath(BLOBSTORE_SUBTREE + key).size();
+    zkClient.close();
+    return replicationCount;
+  }
+
+  @Override
+  public int updateBlobReplication(String key, int replication, Subject who) throws AuthorizationException, KeyNotFoundException {
+    int replicationCount = 0;
+    validateKey(key);
+    SettableBlobMeta meta = getStoredBlobMeta(key);
+    _aclHandler.hasPermissions(meta.get_acl(), WRITE, who, key);
+    try {
+      LOG.warn ("For local file system blob store the update blobs function does not work." +
+                "Please use HDFS blob store to make this feature available. The replication your " +
+                "are noticing is the present replication of the blob based on its availability on various nimbuses");
```

Comment:
Should we instead throw UnsupportedOperationException? This message will go to the nimbus log, and if the user were to attempt to call this, it would appear to have no effect without reason.